### ISSUE TITLE
action missing if state is undefined and dispatch in event handler

### ISSUE THREAD

**Issue by boyangwang** [gaearon/redux-devtools#320]
Hi, this seems to be a bug of redux-devtools. When state is undefined, and dispatch is called from a DOM event handler, the dispatched action will be missing from devtools (it's received by store!)

See below minimal repro:

```
<!DOCTYPE html>
<html>
    <head>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/redux/3.6.0/redux.js"></script>
    </head>
    <body style="height: 200px">
        <script>
            var store = Redux.createStore(function(prev, action) {
                    return prev;
                }, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
            store.subscribe(function() { console.log('xxxx events', store.getState()); });
            store.dispatch({type: 'test1 I will show in devtools no problem'});
            document.body.addEventListener('click', function(e) { store.dispatch({type: 'I do not show up in devtools! why? I am dispatched'}); });
        </script>
    </body>
</html>
```

**Comment by zalmoxisus** [comment]
@boyangwang, yes, it happens for the extension, will fix it. Thanks for the report.

**Comment by boyangwang** [comment]
@zalmoxisus should this issue be opened on [redux-devtools-extension ](https://github.com/zalmoxisus/redux-devtools-extension) and be closed here?

If you may, please point me to some directions and will be happy to submit a PR

**Comment by zalmoxisus** [comment]
@boyangwang, should be fixed in https://github.com/zalmoxisus/remotedev-app/commit/2ab1b1cb79b7bf61a155a5e5ca84450573f257f3. Will publish on Chrome Store by tomorrow.

**Comment by zalmoxisus** [comment]
Fixed in `redux-devtools-extension@2.8.1`. Feel free to reopen the issue if something else goes wrong.
